From 211c196a86980ebb40096d5f8df55d9f973a920b Mon Sep 17 00:00:00 2001
From: Christopher Chavez <chrischavez@gmx.us>
Date: Thu, 11 Nov 2021 09:45:16 -0600
Subject: [PATCH] More accurate check for xdr_sizeof() on macOS

xdr_sizeof() availability depends on the version of macOS,
not the compiler. xdr_sizeof() was provided by a system library
(libsystem_info.dylib) prior to 10.14.
---
 plugin/group_replication/libmysqlgcs/cmake/rpcgen.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugin/group_replication/libmysqlgcs/cmake/rpcgen.cmake b/plugin/group_replication/libmysqlgcs/cmake/rpcgen.cmake
index 62888af52e7e..b3ea14f6a52e 100644
--- a/plugin/group_replication/libmysqlgcs/cmake/rpcgen.cmake
+++ b/plugin/group_replication/libmysqlgcs/cmake/rpcgen.cmake
@@ -43,8 +43,8 @@ IF(WIN32)
 ENDIF()
 
 IF(APPLE)
-  # OSX missing xdr_sizeof() since clang 10.0.1.10010046
-  IF(NOT CMAKE_C_COMPILER_VERSION VERSION_LESS "10.0.1.10010046")
+  # OSX missing xdr_sizeof() since 10.14 Mojave
+  IF(NOT CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS "10.14")
 
     SET(SUNRPC_SRCS
       ${XCOM_BASEDIR}/windeps/sunrpc/xdr_sizeof.c
